docs/data/tree-view/rich-tree-view/ordering/OnlyReorderFromDragHandle.js

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import DragIndicatorIcon from '@mui/icons-material/DragIndicator';
+import { RichTreeViewPro } from '@mui/x-tree-view-pro/RichTreeViewPro';
+
+import { unstable_useTreeItem2 as useTreeItem2 } from '@mui/x-tree-view/useTreeItem2';
+import {
+  TreeItem2Content,
+  TreeItem2IconContainer,
+  TreeItem2GroupTransition,
+  TreeItem2Label,
+  TreeItem2Root,
+  TreeItem2Checkbox,
+} from '@mui/x-tree-view/TreeItem2';
+import { TreeItem2Icon } from '@mui/x-tree-view/TreeItem2Icon';
+import { TreeItem2Provider } from '@mui/x-tree-view/TreeItem2Provider';
+import { TreeItem2DragAndDropOverlay } from '@mui/x-tree-view/TreeItem2DragAndDropOverlay';
+
+const ITEMS = [
+  {
+    id: 'grid',
+    label: 'Data Grid',
+    children: [
+      { id: 'grid-community', label: '@mui/x-data-grid' },
+      { id: 'grid-pro', label: '@mui/x-data-grid-pro' },
+      { id: 'grid-premium', label: '@mui/x-data-grid-premium' },
+    ],
+  },
+  {
+    id: 'pickers',
+    label: 'Date and Time Pickers',
+    children: [
+      { id: 'pickers-community', label: '@mui/x-date-pickers' },
+      { id: 'pickers-pro', label: '@mui/x-date-pickers-pro' },
+    ],
+  },
+];
+
+const CustomTreeItem = React.forwardRef(function CustomTreeItem(props, ref) {
+  const { id, itemId, label, disabled, children, ...other } = props;
+
+  const {
+    getRootProps,
+    getContentProps,
+    getIconContainerProps,
+    getCheckboxProps,
+    getLabelProps,
+    getGroupTransitionProps,
+    getDragAndDropOverlayProps,
+    status,
+  } = useTreeItem2({ id, itemId, children, label, disabled, rootRef: ref });
+
+  const { draggable, onDragStart, onDragOver, onDragEnd, ...otherRootProps } =
+    getRootProps(other);
+
+  const handleDragStart = (event) => {
+    if (!onDragStart) {
+      return;
+    }
+
+    onDragStart(event);
+    event.dataTransfer.setDragImage(event.target.parentElement, 0, 0);
+  };
+
+  return (
+    <TreeItem2Provider itemId={itemId}>
```

Comment:
I don't think we can skip re-creating the whole Tree Item here in order to get the content props and re-dispatch them to the drag handle.

One way of doing it would be to add `handleDragStart` and `handleDragStop` on `useTreeItem2Utils`.
That way, this demo could just override its `content` slot to render the default `TreeItem2Content` with the `children` it receives plus the added icon (and pass the handlers to it from `useTreeItem2Utils`).
It could then use `slotProps` and `defaultMuiPrevented` to remove the behaviors from the root.

Basically it would be closer to [this demo](https://mui.com/x/react-tree-view/rich-tree-view/customization/#limit-expansion-to-icon-container), which would be super nice because if in the future we add editing, or anything else, it would just work out of the box.

But adding it to `useTreeItem2Utils` is not trivial because it means this hook must be able to be enriched from the plugins somehow, and right now it does not.
That will be a nice follow up.

Reply:
Makes a lot of sense to me. I think the DX is good for now and we can think about a follow-up for the utils